text/0000-edition-2021.md

```diff
@@ -0,0 +1,226 @@
+- Feature Name: edition_2021
+- Start Date: 2021-02-19
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC describes the plan for the 2021 Edition. It supercedes [RFC 2052]. The proposed 2021 Edition is intentionally more limited than the 2018 Edition. Rather than representing a major marketing push, the 2021 Edition represents a chance for us to introduce changes that would otherwise be backwards incompatible. It is meant to be marketed in many ways as something closer to "just another release".
+
+Key points include:
+
+* Editions are used to introduce changes into the language that would otherwise have the potential to break existing code, such as the introduction of a new keyword.
+* Editions are never allowed to split the ecosystem. We only permit changes that still allow crates in different editions to interoperate.
+* Editions are named after the year in which they occur (e.g., Rust 2015, Rust 2018, Rust 2021).
+* When we release a new edition, we also release tooling to automate the migration of crates. Some manual work may be required but that should be the rare case.
+* The nightly toolchain offers "preview" access to upcoming editions, so that we can land work that targets future editions at any time.
+* We maintain an Edition Migration Guide that offers guidance on how to migrate to the next edition.
+* Whenever possible, new features should be made to work across all editions.
+
+This RFC is meant to establish the high-level purpose of an edition and to describe how the RFC feels to end users. It intentionally avoids detailed policy discussions, which are deferred to the appropriate subteams (compiler, lang, dev-tools, etc) to figure out.
+
+[RFC 2052]: https://github.com/rust-lang/rfcs/blob/master/text/2052-epochs.md
+
+# Motivation
+[motivation]: #motivation
+
+The plan for editions was laid out in [RFC 2052] and Rust had its first edition in 2018. This effort was in many ways a success but also resulted in some difficult lessons. This RFC proposes a different model for the 2021 Edition. Depending on our experience, we may opt to continue with this model in the future, or we may wish to make changes for future editions.
```

Comment:
What were the lessons? Could some examples be given (very briefly)? So far the only one I saw mentioned was in the alternatives section about 2018 being stressful on org and confusing to some users.

Reply:
Those are two lessons :)

Reply:
That said, I think that primarily I was thinking about the stress on the org. I'll just say that.